database installation/ownership

spencersevilla opened the pull request:
This is a somewhat big feature-add, and I would appreciate some eyes on it.

The major impetus for this change is that haulage is now the only component that uses the mysql database (since we ported colte from OAI to open5gs) so we are now putting haulage in charge of database installation, etc. Changes:

- on installation, postinst script adds a mysql database (default name/pass/uname "haulage_db") with customers, static_ips, and all other tables.

- on installation, we also add a tool ("haulagedb") that lets you add/remove users and top-up their DATA balance (not $). This is just a general system arch choice because haulage does not interface with money, just data.

- I removed some systemd commands from fpm, half because they induced unwanted and IMO incorrect behavior ("restart after upgrade" turns on and enables haulage even if it was turned off) and half because they generated a complicated postinst script that conflicted with what i had written.

- I added a Makefile target (quickstart_ubuntu) that downloads and installs go, mostly because I'm always bringing up new VMs and hate seeking through the download instructions and Ubuntu 18.04 currently ships with go 1.10 so apt doesnt get us there. This script works fine on a clean install of Ubuntu, not sure if it creates issues with other versions of go. Not meant to be 100% authoritative or referenced in the docs, just to help me (and others) out a bit.

matt9j commented on Makefile:
```diff
@@ -35,10 +35,10 @@ package: build
 		--url "https://github.com/uw-ictd/haulage" \
 		--deb-build-depends libpcap-dev \
 		--deb-compression gz \
-		--deb-systemd ./init/haulage.service \
```
Why was this needed? Does it have to do with preinstall/postinstall ordering?

matt9j replied:
Ah, I see the justification is in the PR. I'm not sure if restart after upgrade is incorrect or not in this case. If there are script conflicts then I'm fine with moving to the more manual install done in this commit.

matt9j commented on init/haulagedb.sh:
```diff
@@ -1,7 +1,7 @@
 
 script=/usr/bin/haulagedb.py
-version=0.9.5
+version=1.0
```
Where did this version come from? Haulage definitely should not be 1.0 yet

matt9j commented on init/haulagedb.sh:
```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
```
I'm curious why the shell script that essentially passes through to python? Is it just carryover from a time before when there was just a shell script? Does the shell wrapper do something that I am missing?

matt9j commented on Makefile:
```diff
@@ -48,4 +48,15 @@ package: build
+	export PATH=$PATH:/usr/local/go/bin
+
+
+sudo apt-get -y install libpcap-dev
```
I was going to say this indent is wrong, but it looks like it's fixed in the next commit : )

matt9j commented on init/haulagedb.py:
```diff
@@ -0,0 +1,127 @@
+#########################################################################
+############### OPTION FOUR: DISABLE A USER (AND ZERO-OUT BALANCE???) ###
+#########################################################################
+# elif (command == "disable"):
```
I don't like the dead code here... if we don't want these functions let's just get rid of them?

matt9j commented on init/postrm:
```diff
@@ -0,0 +1,10 @@
+	echo "Purged haulage_db user and database"
+fi
+
+echo "Uninstalled haulage"
```
Do we need to make any systemd calls as part of postrm now that we're not using FPM's systemd integration?

spencersevilla commented:
thanks for all these comments! super helpful, have addressed as follows:

- using version 0.9.6 for now
- shell script was a holdover from an olden time. changed to just the .py file with "#!/usr/bin/env python3" at the top
- dead code removed, its easy to write again if we want
- added systemd stop and daemon-reload commands to postrm.
